guess_the_words/function.py

```diff
@@ -0,0 +1,161 @@
+
+
+def val_count(player_letters: list) -> list:
+    if len(player_letters) > 5:
+        print('You enter word more then 5 letters')
+        return True
+    elif len(player_letters) < 5:
+        print('You enter word less then 5 letters')
+        return True
+
+
+def rules_for_user():
+    with open('Rules.txt', 'r') as file:
+        open_file = file.read()
+    return open_file
+
+
+def hello_choose(player_name: str) -> str:
+    choose_eser = input(f'Hello, {player_name}. Choose what you need:'
+                        f'\n1. Read rules of game - type "rules"'
+                        f'\n2. To start game - type "start"'
+                        f'\n3. To stop game - type "stop"'
+                        f'\n4. Show past result - type "result"')
+    return choose_eser
+
+
+def search_result(name_player: str) -> str:
+    with open('guess_result.json', 'r') as file:
+        data = json.load(file)
+        list_data = data['Result of games']
+        ii = 0
+        while ii < len(list_data):
+            list_j = []
+            ii += 1
+            for i, j in enumerate(list_data):
+                if name_player == j['Player name']:
+                    list_j.append(j)
+            return list_j
+
+
+def choose_loop(choose_user: str, player_name: str) -> str:
+    while True:
+        if choose_user == 'rules'.lower():
+            rules = rules_for_user()
+            print(rules)
+            break
+        elif choose_user == 'stop'.lower():
+            print('Game was stopped')
+            break
+        elif choose_user == 'start'.lower():
+            body_loop(player_name)
+            break
+        elif choose_user == 'result'.lower():
+            name_player = input('Enter player name: ').title()
+            result_search = search_result(name_player)
+            if result_search == []:
+                print('Player not defined. Try another name!')
+                break
+            else:
+                print(result_search)
+                break
+
+        else:
+            print('Incorrect input.')
+            break
+
+
+def val_alpha(player_letters: list) -> list:
+    if set(player_letters) - set(string.ascii_lowercase):
+        print("Unsupported symbols, use letters instead")
+        return True
+
+
+def result_add(count_dict):
+    with open('guess_result.json', 'r') as file:
+        data = json.load(file)
+        list_data = data['Result of games']
+        add_list = count_dict
+        list_data.append(add_list)
+    with open('guess_result.json', 'w') as file:
+        json.dump(data, file, indent=4)
+
+
+def body_loop(player_name):
+    count_dict = {'Date time game': '', 'Player name': player_name, 'Game result': 'Lose',
+                  'Score': {'Count tries': 0, 'Score points': 0}}
+
+    target_word = random.choice(create_list())
+
+    i = 0
+    while i < 6:
+        player_letters = list(input(f'Attempt {i + 1} of 6: ').lower())
+
+        if val_alpha(player_letters):
```

Comment:
You can skip this for now, but in future it's better to make something like that:
```
error = val_alpha(player_letters)
if error is not None:
    print(error)
    continue
```

It's better to understand waht function does (it return an error string in case of the word is invalid) and does not print enything:
```
def val_count(player_letters: list) -> Optional[str]:
    if len(player_letters) > 5:
        return 'You enter word more then 5 letters'

    if len(player_letters) < 5:
        return 'You enter word less then 5 letters'

    return None
```

Reply:
So far, I have removed two functions - I discovered a bug in which, if you enter less than five letters, a message is written that you entered less than 5 letters and the program shows whether they were guessed.
Replaced with the code.